GUI.java

```diff
@@ -0,0 +1,221 @@
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+
+class GUIForm{ //one form stores one JPanel at a time
+    ArrayList<JLabel> keyList = new ArrayList<JLabel>();
+    ArrayList<JTextField> valueList = new ArrayList<JTextField>();
+    JPanel pnForm = new JPanel();
+    int itemCounter=0; //gotta know how many items we have
+
+    //this method accepts a string full of key values, creates a list of form elements,
+    //then sets up a panel which can then be used to render into any JOptionPane later
+    public GUIForm(String _keyString //a string with key strings seperated by semicolon
+                    // eg. "product;quantity;wholesalePrice;...;" also has to end with a semicolon
+            , int _fieldSize){
+        String output="",str="";
+        for(int i=0;i<_keyString.length();i++){
+            str=_keyString.substring(i,i+1); //check for ";"
+            if (str.equals(";")){
+                itemCounter++;
+                keyList.add(new JLabel(output));
+                output="";
+                valueList.add(new JTextField(_fieldSize));
+                continue; //to prevent inclusion of semicolon in our output string
+            }
+            output=output+str;
+        }
+        //now create an appropriate jpanel to house those form elements
+
+        pnForm.setLayout(new GridLayout(itemCounter,2));
+        for (int i=0;i<keyList.size();i++){
+            pnForm.add(keyList.get(i));
+            pnForm.add(valueList.get(i));
+        }// add the keys and values one by one in a grid style
+        //JPanel is ready for deployment!!
+    }
+
+    String getKeyValue(String _key){
+        for(int i=0;i<keyList.size();i++){
+            if (keyList.get(i).getText().equals(_key)){
+                return valueList.get(i).getText();
+            }
+        }
+        return "key not found";
+    }
+    void showFormDialog(String _title){
+        for (int i=0;i<valueList.size();i++){
+            valueList.get(i).setText("");
+        } //clear the form from previous entry
+        JOptionPane.showMessageDialog(null,pnForm,_title,JOptionPane.INFORMATION_MESSAGE);
+    }
+}
+
+class Screen{
+    JPanel east = new JPanel();
+    JPanel west = new JPanel();
+    JPanel north = new JPanel();
+    JPanel south = new JPanel();
+    JPanel center = new JPanel();
+    JPanel bottom = new JPanel();
+
+    JFrame frame = new JFrame();
+
+    public Screen() { //does this constructor have to be public??
```

Comment:
i would think not? nothing would be calling the screen outside of this class correct?